src/core/event.js

```diff
@@ -482,9 +482,19 @@ Candy.Core.Event = (function(self, Strophe, $) {
 			}
 
 			if(directInvite.length > 0) {
+				fromUser = Candy.Core.getRoster().get(msg.attr('from'));
+				room = Candy.Core.getRoom(directInvite.attr('jid'));
```

Comment:
If this is an invite, we won't have the room object yet - is it useful to pre-load it into the collection (potential memory leak) or do we just go back to passing the JID forward (perhaps as a JID object rather than a string)?